### PR TITLE
fix temperature flag so it takes effect

### PR DESCRIPTION
**Pull request by RaviTriv**
Currently the temperature flag is not taking effect since its applied to the exponentiated logits and cancels against the normalization. Updating to divide the logits by temperature before exp so it affects the output.

## Current
**Temperature 0**
image

**Temperature 2**
image

## Updated
**Temperature 0**
image

**Temperature 2**
image

image

### CODE REVIEW

File: ops/ops-inl.h [hunk 1]
```diff
@@ -1352,6 +1350,7 @@ static HWY_NOINLINE void Softmax(Logits logits, ThreadingContext& ctx,
                                  const SMOptions& sm_options = {}) {
   GCPP_ZONE(ctx, worker, Zones::kOpsSoftmax);
   HWY_DASSERT(logits.size() != 0);
+  HWY_ASSERT(temperature >= 0.0f);
```
Comment (jan-wassenberg) [inline ops/ops-inl.h]: This runs quite often, so prefer HWY_DASSERT.

**Review by jan-wassenberg — CHANGES_REQUESTED** [review]
Thanks for fixing. We generally run with deterministic mode because our sampler is very basic, but good to fix this. One nit:

File: ops/ops-inl.h (hunk 1 above)
Comment (RaviTriv) [reply ops/ops-inl.h]: good catch

**Review by jan-wassenberg — APPROVED** [review]
